compiler/il/AliasSetInterface.hpp

```diff
@@ -36,19 +36,31 @@
 #include "infra/Assert.hpp"
 #include "infra/BitVector.hpp"
 
+typedef enum {
+   useDefAliasSet,
+   UseOnlyAliasSet
+} AliasSetInterface;
 
-template <class AliasSetInterface>
+template <uint32_t _AliasSetInterface>
```

Comment:
why is this template parameter a uint32_t rather than having a full enum and using the enum here since this only ever seems to be compared with the enum - is there a compiler limitation or something I am missing?

Reply:
Since there are forward declarations of this class, using `uint32_t ` would be more convenient as we don't need to define the enum every time.

Reply:
can't you just forward declare the enum as well?

Reply:
I think it is possible to also forward declare the enum, but it would make it less convenient whenever we want to use the class `TR_AliasSetInterface`. Do you think we should still do that?

Reply:
I have changed the template parameter to enum `AliasSetInterface` in c72f7b1.